Common/src/Storage/TaskLifeCycleHelper.cs

```diff
@@ -476,10 +485,22 @@ await pushQueueStorage.PushMessagesAsync(group,
 
     await taskTable.FinalizeTaskCreation(messages.Select(task => task.TaskId)
                                                  .AsICollection(),
+                                         sessionData.Status == SessionStatus.Paused,
                                          cancellationToken)
                    .ConfigureAwait(false);
   }
 
+  /// <summary>
+  ///   Resume session and its paused tasks
+  /// </summary>
+  /// <param name="taskTable">Interface to manage task states</param>
+  /// <param name="sessionTable">Interface to manage session states</param>
+  /// <param name="pushQueueStorage">Interface to push tasks in the queue</param>
+  /// <param name="sessionId">Id of the session to resume</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   The updated data of the session
+  /// </returns>
```

Comment:
The tasks are pushed to the queue and then their status is updated. A race condition could occur if some polling agent pulls a task from the queue before the said task is switched to "submitted". The polling agent would then remove it off the queue without putting it back